### PR TITLE
[CAMEL-18618] Milo component: Return status from write

### PR DESCRIPTION
**Pull request by rangoy**
- [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [X] Each commit in the pull request should have a meaningful subject line and body.
- [X] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] ? Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/main/CONTRIBUTING.md

In our usecase we need to get the response from the server when writing values to OPC UA via Milo. 

I've tried to ask on [zulipchat|https://camel.zulipchat.com/#narrow/stream/257298-camel/topic/Result.20from.20camel-milo.20endpoint] and also create an issue on jira without any response.

I tried to run a clean build in the component directory but tests failed both on master and my branch

### CODE REVIEW

**Comment by davsclaus** [comment]
At first glance this seems a bit wrong as future.get() is a blocking operation. And the method returns false which is signalling to camel that the response is async and completes later.

So for me it looks the old code, in that whenComplete you then need to deal with exception and set on the exchange before calling async.done(false)

**Comment by rangoy** [comment]
Hi, thanks for the reply. 

As I said, when I asked at [zulipchat](https://camel.zulipchat.com/#narrow/stream/257298-camel/topic/Result.20from.20camel-milo.20endpoint/near/300812919)  , I was not sure how to do that. Do you have any suggestion how to do that?

> I did an attempt of patching this to be able to get the status from the server, not sure how to get the new body applied without using get()

**Comment by davsclaus** [comment]
I would assume future get is available during whenComplete is called

**Comment by rangoy** [comment]
> I would assume future get is available during whenComplete is called

I finally got to this again. For some reason I did not get that to work last time, but this seems to work as intended. I guess parsing of the status value can be left for the consumer. 

```
  future.whenComplete((o, throwable) -> {
                        final DefaultMessage newMessage = new DefaultMessage(exchange);
                        newMessage.setBody(o);
                        exchange.setMessage(newMessage);
                        async.done(false);
                    }
            );
            return false;
```

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java [hunk 1]
```diff
@@ -91,7 +92,13 @@ public boolean process(Exchange exchange, AsyncCallback async) {
         final Boolean await = msg.getHeader(MiloConstants.HEADER_AWAIT, this.defaultAwaitWrites, Boolean.class);
 
         if (TRUE.equals(await)) {
-            future.whenComplete((v, ex) -> async.done(false));
+            future.whenComplete((o, throwable) -> {
+                        final DefaultMessage newMessage = new DefaultMessage(exchange);
```
Comment (davsclaus) [inline components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java:96]: I dont think you need to set a new message as this leads to loosing headers and whatnot.

You usually just set the result directly on the message, eg

exchange.getMessage().setBody(...)

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import org.apache.camel.Message;
 import org.apache.camel.component.milo.MiloConstants;
 import org.apache.camel.support.DefaultAsyncProducer;
+import org.apache.camel.support.DefaultMessage;
```
Comment (davsclaus) [inline components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java:28]: Remove this unusued import

**Review by davsclaus — APPROVED** [review]

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java [hunk 3]
```diff
@@ -91,7 +92,11 @@ public boolean process(Exchange exchange, AsyncCallback async) {
         final Boolean await = msg.getHeader(MiloConstants.HEADER_AWAIT, this.defaultAwaitWrites, Boolean.class);
 
         if (TRUE.equals(await)) {
-            future.whenComplete((v, ex) -> async.done(false));
+            future.whenComplete((result, throwable) -> {
+                        msg.setBody(result);
```
Comment (davsclaus) [inline components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java:96]: Ah what if throwable != null, eg I assume there was an error, then its likely better to set response, on the exchange

if (throwable != null) {
msg.getExchange().setException(throwable);
} else {
  msg.setBody(result);
}

Comment (rangoy) [reply components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java]: Ok, I amended the commit where I added that and also removed an unnecessary import also.

**Review by davsclaus — APPROVED** [review]

**Comment by rangoy** [comment]
Rebased onto latest main

**Pull request merged**